knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -78,7 +86,130 @@ func (r *ReconcileKnativeKafka) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
-	// NOT IMPLEMENTED YET
+	// TODO: check for deletion
+	// if original.GetDeletionTimestamp() != nil {
+	//		return reconcile.Result{}, r.delete(original)
+	//	}
+
+	instance := original.DeepCopy()
+	reconcileErr := r.reconcileKnativeKafka(instance)
+
+	if !equality.Semantic.DeepEqual(original.Status, instance.Status) {
+		if err := r.client.Status().Update(context.TODO(), instance); err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to update status: %w", err)
+		}
+	}
+
+	if instance.Status.IsReady() {
+		common.KnativeKafkaUpG.Set(1)
+	} else {
+		common.KnativeKafkaUpG.Set(0)
+	}
+	return reconcile.Result{}, reconcileErr
+}
+
+func (r *ReconcileKnativeKafka) reconcileKnativeKafka(instance *operatorv1alpha1.KnativeKafka) error {
+	stages := []func(*operatorv1alpha1.KnativeKafka) error{
+		// TODO r.configure,
+		// TODO r.ensureFinalizers,
```

Comment:
As per comment above (https://github.com/openshift-knative/serverless-operator/pull/487/files#r496717753), no namespace injection nor finalizer setting is done yet. Will be implemented in a separate PR.